lib/tasks/buildThemes.js

```diff
@@ -128,19 +164,51 @@ export default async function({
 				log.verbose(`Available sap.ui.core themes: ${availableThemes.join(", ")}`);
 			}
 		}
-		allResources = allResources.filter(isAvailable);
+		themeResources = themeResources.filter(isAvailable);
 	}
 
-	const processedResources = await themeBuilder({
-		resources: allResources,
-		fs: fsInterface(combo),
-		options: {
-			compress,
-			cssVariables: !!cssVariables
+	let buildTheme;
+	if (useWorkers) {
+		if (!taskUtil) {
+			// TaskUtil is required for worker support
+			throw new Error(`buildThemes: Option 'useWorkers' requires a taskUtil instance to be provided`);
 		}
-	});
+		buildTheme = buildThemeInWorker;
+	} else {
+		// Do not use workerpool
+		buildTheme = (await import("../processors/themeBuilderWorker.js")).default;
+	}
+
+	// All the relevant resources need to be made available for the less parser within the workerpool
```

Comment:
All those things are not relevant when `useWorkers=false`. I'd expect that the new handling is only applied when using workers. Otherwise it might cause unnecessary overhead for existing use cases.

Reply:
Done